lib/libc/powerpc64/string/strcmp.S

```diff
@@ -34,164 +34,121 @@ __FBSDID("$FreeBSD$");
 ENTRY(strcmp)
 
 .Lstrcmp_loop:
-        // Load double words.
+        /* Load double words. */
         ld %r5,0(%r3)
         ld %r6,0(%r4)
 
-        // Check if double words contain different chars.
-        cmpd  %r5,%r6
-        bne .Lstrcmp_check_differences
+        /* Check if double words are different. */
+        cmpb %r0,%r5,%r6
```

Comment:
You can replace cmpb + not (nand) with a single xor %r0, %r5, %r6

Reply:
That works too. Thanks. I will change that.